.github/workflows/_ci.yml

```diff
@@ -100,7 +99,13 @@ jobs:
       with:
         image_tag: ${{ inputs.cache_build_image }}
 
-    - if: ${{ inputs.trusted && secrets.app_id && secrets.app_key }}
+    - name: Check workflow context
+      id: context
+      run: |
+        if [[ "${{ inputs.trusted }}" != "false" && -n "${{ secrets.app_id }}" && -n "${{ secrets.app_key }}" ]]; then
+            echo "use_appauth=true" >> $GITHUB_OUTPUT
+        fi
```

Comment:
I am still a little worry about this... But let's land it first and check it again.